src/DeepCopy/TypeMatcher/Doctrine/DoctrineProxyNotCloneableTypeMatcher.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace DeepCopy\TypeMatcher\Doctrine;
+
+use DeepCopy\TypeMatcher\TypeMatcherInterface;
+use Doctrine\Common\Persistence\Proxy;
+use ReflectionObject;
+
+class DoctrineProxyNotCloneableTypeMatcher implements TypeMatcherInterface
```

Comment:
Why is the cloneable check even here? imo it's not the matcher's resposibility.

Moving this out of the class would also clear the name to 'DoctrineProxyTypeMatcher'